render/renderable_node.go

```diff
@@ -8,14 +8,15 @@ import (
 // an element of a topology. It should contain information that's relevant
 // to rendering a node when there are many nodes visible at once.
 type RenderableNode struct {
-	ID         string                   `json:"id"`                    //
-	LabelMajor string                   `json:"label_major"`           // e.g. "process", human-readable
-	LabelMinor string                   `json:"label_minor,omitempty"` // e.g. "hostname", human-readable, optional
-	Rank       string                   `json:"rank"`                  // to help the layout engine
-	Pseudo     bool                     `json:"pseudo,omitempty"`      // sort-of a placeholder node, for rendering purposes
-	Adjacency  report.IDList            `json:"adjacency,omitempty"`   // Node IDs (in the same topology domain)
-	Origins    report.IDList            `json:"origins,omitempty"`     // Core node IDs that contributed information
-	Metadata   report.AggregateMetadata `json:"metadata"`              // Numeric sums
+	ID           string                   `json:"id"`                    //
+	LabelMajor   string                   `json:"label_major"`           // e.g. "process", human-readable
+	LabelMinor   string                   `json:"label_minor,omitempty"` // e.g. "hostname", human-readable, optional
+	Rank         string                   `json:"rank"`                  // to help the layout engine
+	Pseudo       bool                     `json:"pseudo,omitempty"`      // sort-of a placeholder node, for rendering purposes
+	Adjacency    report.IDList            `json:"adjacency,omitempty"`   // Node IDs (in the same topology domain)
+	Origins      report.IDList            `json:"origins,omitempty"`     // Core node IDs that contributed information
+	Metadata     report.AggregateMetadata `json:"metadata"`              // Numeric sums
+	NodeMetadata report.NodeMetadata      `json:"-"`                     // merged NodeMetadata of the nodes used to build this
```

Comment:
I think it'd be good to disambiguate the two metadatas by e.g.

```go
type RenderableNode struct {
    // ...
    AggregateMetadata report.AggregateMetadata `json:"metadata"`
    NodeMetadata     report.NodeMetadata `json:"-"`
}
```

or, perhaps even better,

```
type RenderableNode struct {
    // ...
    report.AggregateMetadata `json:"metadata"`
    report.NodeMetadata      `json:"-"`
}
```